### PR TITLE
feat: add ontask service definitions for dev/local

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds the necessary definitions for the OnTask service: docker-compose files, initialization tasks, configurations, and variables. 

### How to test
1. Install and enable this plugin in your tutor environment using the `MJG/ondata-service` branch
2. Initialize services: `tutor dev|local do init --limit ontask`
3. Run services: `tutor dev|local start -d ontask`
4. Check your OnTask installation: `ontask.<LMS DOMAIN>` 
image

### CODE REVIEW

File: tutorontask/templates/ontask/build/ontask/Dockerfile [hunk 1]
```diff
@@ -22,20 +20,74 @@ RUN apt-get update && apt-get upgrade -y \
...
+# Install Python
+RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
+
+# Create virtualenv
+RUN $PYENV_ROOT/versions/$PYTHON_VERSION/bin/python -m venv /app/venv
+
+###### Checkout OnTask code
+
+RUN mkdir -p /app/ontask && \
+    git clone https://github.com/abelardopardo/ontask_b.git --branch Version_10_3 --depth 1 /app/ontask
```
Comment (mariajgrimaldi) [inline tutorontask/templates/ontask/build/ontask/Dockerfile]:
```suggestion
    git clone https://github.com/abelardopardo/ontask_b.git --branch {{ ONTASK_APP_VERSION }} --depth 1 /app/ontask
```

File: tutorontask/templates/ontask/jobs/init/ontask/init.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+>&2 echo "Creating initial data (learners)"
+./manage.py initialize_db /app/ontask/scripts/initial_learners.csv
+
+>&2 echo "Creating superuser "
+./manage.py create_superuser -u $SUPERUSER_NAME -e $SUPERUSER_EMAIL -p $SUPERUSER_PWD
+
+# Collect static
+>&2 echo "Collecting static"
+./manage.py collectstatic --noinput
```
Comment (mariajgrimaldi) [inline tutorontask/templates/ontask/jobs/init/ontask/init.sh]: Remove these lines

File: tutorontask/templates/ontask/build/ontask/Dockerfile [hunk 3]
```diff
@@ -22,20 +20,74 @@ RUN apt-get update && apt-get upgrade -y \
...
+
+# Install python with pyenv in /opt/pyenv and create virtualenv in /ontask/venv
+# https://github.com/pyenv/pyenv/wiki/Common-build-problems#prerequisites
+RUN apt update && \
+    apt install -y libssl-dev zlib1g-dev libbz2-dev \
+    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+    xz-utils tk-dev libffi-dev liblzma-dev python3-openssl git
+
+# Install apache2
+RUN apt-get update && apt-get install -y apache2 apache2-utils libapache2-mod-wsgi-py3
```
Comment (BryanttV) [inline tutorontask/templates/ontask/build/ontask/Dockerfile]: Why install apache2?

File: tutorontask/templates/ontask/jobs/init/ontask/init.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+./manage.py migrate --noinput
+
+>&2 echo "Creating initial data (instructors)"
+./manage.py initialize_db -i /app/ontask/scripts/initial_instructors.csv
+
+>&2 echo "Creating initial data (learners)"
+./manage.py initialize_db /app/ontask/scripts/initial_learners.csv
+
+>&2 echo "Creating superuser "
+./manage.py create_superuser -u $SUPERUSER_NAME -e $SUPERUSER_EMAIL -p $SUPERUSER_PWD
```
Comment (BryanttV) [inline tutorontask/templates/ontask/jobs/init/ontask/init.sh]:
```suggestion
./manage.py create_superuser -u {{ ONTASK_SUPERUSER_NAME }} -e {{ ONTASK_SUPERUSER_EMAIL }} -p {{ ONTASK_SUPERUSER_PWD }}

```

Comment (mariajgrimaldi) [reply tutorontask/templates/ontask/jobs/init/ontask/init.sh]: thanks!

File: tutorontask/templates/ontask/build/ontask/Dockerfile (hunk 3 above)
Comment (mariajgrimaldi) [reply tutorontask/templates/ontask/build/ontask/Dockerfile]: I should remove this! Sorry

File: tutorontask/plugin.py [hunk 5]
```diff
@@ -20,6 +20,14 @@
         # Each new setting is a pair: (setting_name, default_value).
         # Prefix your setting names with 'ONTASK_'.
         ("ONTASK_VERSION", __version__),
+        ("ONTASK_HOST", "ontask.{{ LMS_HOST }}"),
+        ("ONTASK_DB_NAME", "ontask"),
+        ("ONTASK_DB_USER", "ontask"),
+        ("ONTASK_POSTGRES_HOST", "postgres"),
+        ("ONTASK_DOCKER_IMAGE", "docker.io/edunext/ontask:0.1.0"),
```
Comment (BryanttV) [inline tutorontask/plugin.py]:
```suggestion
        ("ONTASK_DOCKER_IMAGE", "docker.io/edunext/ontask:0.1.1"),
```

**Review by BryanttV — APPROVED** [review]

**Pull request merged**